docs/src/main/asciidoc/mongodb-panache-guide.adoc

```diff
@@ -0,0 +1,505 @@
+
+    public String setName(String name){
+      this.name = name.toLowercase();
+    }
+}
+--
+
+And thanks to our field access rewrite, when your users read `person.name` they will actually call your `getName()` accessor, and similarly for field writes and the setter.
+This allows for proper encapsulation at runtime as all fields calls will be replaced by the corresponding getter/setter calls.
+
+== Most useful operations
+
+Once you have written your entity, here are the most common operations you will be able to do:
+
+[source,java]
+--
+// creating a person
+Person person = new Person();
+person.name = "Stef";
+person.birth = LocalDate.of(1910, Month.FEBRUARY, 1);
+person.status = Status.Alive;
+
+// persist it
+person.persist();
+
+// delete it
+person.delete();
+
+// getting a list of all Person entities
+List<Person> allPersons = Person.listAll();
+
+// finding a specific person by ID
+person = Person.findById(personId);
+
+// finding all living persons
+List<Person> livingPersons = Person.list("status", Status.Alive);
+
+// counting all persons
+long countAll = Person.count();
+
+// counting all living persons
+long countAlive = Person.count("status", Status.Alive);
+
+// delete all living persons
+Person.delete("status", Status.Alive);
+
+// delete all persons
+Person.deleteAll();
+--
+
+All `list` methods have equivalent `stream` versions.
+
+[source,java]
+--
+Stream<Person> persons = Person.streamAll();
+List<String> namesButEmmanuels = persons
+    .map(p -> p.name.toLowerCase() )
+    .filter( n -> ! "emmanuel".equals(n) )
+    .collect(Collectors.toList());
+--
+
+
+== Paging
+
+You should only use `list` and `stream` methods if your collection contains small enough data sets. For larger data
+sets you can use the `find` method equivalents, which return a `PanacheQuery` on which you can do paging:
+
+[source,java]
+--
+// create a query for all living persons
+PanacheQuery<Person> livingPersons = Person.find("status", Status.Alive);
+
+// specify a projection to limit the fields returned from the database
+livingPersons.project("name", "birthdate");
+
+// make it use pages of 25 entries at a time
+livingPersons.page(Page.ofSize(25));
+
+// get the first page
+List<Person> firstPage = livingPersons.list();
+
+// get the second page
+List<Person> secondPage = livingPersons.nextPage().list();
+
+// get page 7
+List<Person> page7 = livingPersons.page(Page.of(7, 25)).list();
+
+// get the number of pages
+int numberOfPages = livingPersons.pageCount();
+
+// get the total number of entities returned by this query without paging
+int count = livingPersons.count();
+
+// and you can chain methods of course
+return Person.find("status", Status.Alive)
+    .page(Page.ofSize(25))
+    .nextPage()
+    .stream()
+--
+
+The `PanacheQuery` type has many other methods to deal with paging and returning streams.
+
+== Sorting
+
+All methods accepting a query string also an optional `Sort` parameter, which allows your to abstract your sorting:
```

Comment:
`All methods accepting a query string also accept an optional `Sort` parameter, which allows your to abstract your sorting`

Reply:
done